PR: https://github.com/odoo/odoo/pull/186906

From: 9144eb051538d5a92d403e1ad9c635ce39e090ec
From: Gorash
Date: 2024-11-14 10:39:02

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = ['pos.config']
-    _inherit = ['pos.order']
-    _inherit = ['res.config.settings']

Total Changes: 6

[REV] *: Restore the model `_name`

Following internal discussions, the python typing and inheritance task is
canceled. The use of the class name instead of '_name' was done to allow
notations such as `partner_id = fields.Many2one[base.ResParter](string='Partner')`.
Since this notation is canceled, we put `_name` back (although the class
name is sufficient).

Partial revert:
https://github.com/odoo/enterprise/pull/69762
https://github.com/odoo/odoo/pull/178200

closes odoo/odoo#186906

Related: odoo/enterprise#73643
Signed-off-by: Raphael Collet <rco@odoo.com>

================================= pseudo patch: =================================

--- a/addons/pos_sms/models/pos_config.py
+++ b/addons/pos_sms/models/pos_config.py
@@ -2,6 +2,6 @@ from odoo import fields, models
 
 
 class PosConfig(models.Model):
-    _inherit = ['pos.config']
+    _inherit = 'pos.config'
 
     sms_receipt_template_id = fields.Many2one('sms.template', string="Sms Receipt template", domain=[('model', '=', 'pos.order')], help="SMS will be sent to the customer based on this template")

--- a/addons/pos_sms/models/pos_order.py
+++ b/addons/pos_sms/models/pos_order.py
@@ -2,7 +2,7 @@ from odoo import models
 
 
 class PosOrder(models.Model):
-    _inherit = ['pos.order']
+    _inherit = 'pos.order'
 
     def action_sent_message_on_sms(self, phone, _):
         if not (self and self.config_id.module_pos_sms and self.config_id.sms_receipt_template_id and phone):

--- a/addons/pos_sms/models/res_config_settings.py
+++ b/addons/pos_sms/models/res_config_settings.py
@@ -2,6 +2,6 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = ['res.config.settings']
+    _inherit = 'res.config.settings'
 
     pos_sms_receipt_template_id = fields.Many2one('sms.template', related='pos_config_id.sms_receipt_template_id', readonly=False)
